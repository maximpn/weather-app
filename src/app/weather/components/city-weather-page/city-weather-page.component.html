<wa-page-header>
  <h1>Hourly weather forecast in {{ cityName }}</h1>
</wa-page-header>

<div class="hourly-forecast" role="list">
  <ng-container *ngFor="let weather of forecast">
    <wa-timestamp-weather [weather]="weather" role="listitem"></wa-timestamp-weather>
    <mat-divider></mat-divider>
  </ng-container>
</div>
