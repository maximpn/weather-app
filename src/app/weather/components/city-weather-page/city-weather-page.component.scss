.hourly-forecast {
  padding: 0 20%;
}
