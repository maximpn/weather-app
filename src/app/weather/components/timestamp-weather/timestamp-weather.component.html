<div class="date-time-wrapper">
  <time class="time">{{ weather.dt * 1000 | date:'H:mm' }}</time>
  <time class="date">{{ weather.dt * 1000 | date:'MMM d' }}</time>
</div>

<wa-weather-condition-icon [weatherCondition]="weather.weather[0]" size="2x"></wa-weather-condition-icon>

<span class="temperature">{{ weather.temp | number:'1.0-0' }}°</span>
<span class="real-feel">Feels like: {{ weather.feels_like | number:'1.0-0' }}°</span>
<span class="phrase">{{ weather.weather[0].description }}</span>
<div class="clouds">
  <mat-icon>cloud_queue</mat-icon>
  <span class="value">{{ weather.clouds }}%</span>
</div>
