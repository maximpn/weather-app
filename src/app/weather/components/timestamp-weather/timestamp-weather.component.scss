:host {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
}

.date-time-wrapper {
  width: 80px;
  display: flex;
  flex-direction: column;
  align-items: center;

  .time {
    font-size: 32px;
  }

  .date {
    margin-top: 5px;
  }
}

.temperature {
  font-size: 48px;
}

.real-feel {
  color: #878787;
}

.phrase {
  width: 200px;
  text-align: center;
  text-transform: capitalize;
}

.clouds {
  display: flex;

  .value {
    line-height: 24px;
    margin-left: 10px;
  }
}
