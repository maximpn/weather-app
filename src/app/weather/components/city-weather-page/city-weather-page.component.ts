import { Component } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

import { TimestampWeather } from '../../models/timestamp-weather.model';

@Component({
  templateUrl: 'city-weather-page.component.html',
  styleUrls: ['city-weather-page.component.scss'],
})
export class CityWeatherPageComponent {
  forecast: TimestampWeather[] | Error = this.route.snapshot.data.forecast;

  constructor(private readonly route: ActivatedRoute) {}

  get cityName(): string {
    return this.route.snapshot.params.name;
  }
}
