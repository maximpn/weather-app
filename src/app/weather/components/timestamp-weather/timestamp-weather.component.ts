import { Component, Input } from '@angular/core';

import { TimestampWeather } from '../../models/timestamp-weather.model';

@Component({
  selector: 'wa-timestamp-weather',
  templateUrl: 'timestamp-weather.component.html',
  styleUrls: ['timestamp-weather.component.scss'],
})
export class TimestampWeatherComponent {
  @Input()
  weather!: TimestampWeather;
}
