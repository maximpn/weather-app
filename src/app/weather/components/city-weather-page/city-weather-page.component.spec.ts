import { Component, Input, NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';

import { TimestampWeather } from '../../models/timestamp-weather.model';

import { CityWeatherPageComponent } from './city-weather-page.component';

@Component({
  selector: 'wa-timestamp-weather',
  template: '{{ weather | json }}',
})
class TimestampWeatherMockComponent {
  @Input()
  weather!: TimestampWeather;
}

describe('CityWeatherPageComponent', () => {
  let fixture: ComponentFixture<CityWeatherPageComponent>;

  beforeEach(() => {
    const activatedRouteMock = {
      snapshot: {
        params: {
          name: 'City1',
        },
        data: {
          forecast: [
            { temp: 20 },
            { temp: 21 },
            { temp: 22 },
          ],
        },
      },
    };

    fixture = TestBed.configureTestingModule({
      declarations: [
        CityWeatherPageComponent,
        TimestampWeatherMockComponent,
      ],
      providers: [
        { provide: ActivatedRoute, useValue: activatedRouteMock },
      ],
      schemas: [
        NO_ERRORS_SCHEMA,
      ],
    }).createComponent(CityWeatherPageComponent);

    fixture.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
