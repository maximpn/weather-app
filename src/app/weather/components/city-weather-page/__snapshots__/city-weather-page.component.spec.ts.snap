// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeatherPageComponent should render the component 1`] = `
<div
  id="root0"
>
  <wa-page-header>
    <h1>
      Hourly weather forecast in City1
    </h1>
  </wa-page-header>
  <div
    class="hourly-forecast"
    role="list"
  >
    
    
    <wa-timestamp-weather
      role="listitem"
    >
      {
  "temp": 20
}
    </wa-timestamp-weather>
    <mat-divider />
    
    <wa-timestamp-weather
      role="listitem"
    >
      {
  "temp": 21
}
    </wa-timestamp-weather>
    <mat-divider />
    
    <wa-timestamp-weather
      role="listitem"
    >
      {
  "temp": 22
}
    </wa-timestamp-weather>
    <mat-divider />
  </div>
</div>
`;
